### PR TITLE
chore(cypress): Fix user profile e2e test

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
Fix: https://github.com/mattermost/mattermost/pull/26420#issuecomment-2113202367

### CODE REVIEW

File: e2e/tests/join_call.spec.ts [hunk 1]
```diff
@@ -109,8 +112,9 @@ test.describe('join call', () => {
         await userBPage.page.locator('[data-testid=SendMessageButton]').click();
 
         await userAPage.locator('.post__header').locator('button.user-popover').last().click();
-        await expect(userAPage.locator('#user-profile-popover')).toBeVisible();
-        await expect(userAPage.locator('#user-profile-popover').locator('#startCallButton')).toBeDisabled();
+        await expect(userAPage.locator('div.user-profile-popover')).toBeVisible();
+        await expect(userAPage.locator('div.user-profile-popover').locator('#startCallButton')).toBeDisabled();
+        await userAPage.locator('button.closeButtonRelativePosition').click();
```
Comment (cpoile) [inline e2e/tests/join_call.spec.ts:117]: Just a thought -- will this be robust if another button is added on the page? Could we limit the locator to the `div.user-profile-popover` to make it more future proof?

**Review by cpoile — APPROVED** [review]
Thanks @yasserfaraazkhan!

**Comment by streamer45** [comment]
@yasserfaraazkhan Given these are now passing even though the functionality is still not completely back (thanks for https://mattermost.atlassian.net/browse/MM-58271), would it make sense to update the test to cover what's missing? Let me know if you need help with that.

File: e2e/tests/join_call.spec.ts (hunk 1 above)
Comment (yasserfaraazkhan) [reply e2e/tests/join_call.spec.ts:117]: @cpoile with the new changes for pop over.
The testing tool encounters some kinda overlay when attempting to interacting with other elements while the profile pop over is open. This `button.closeButtonRelativePosition` is just a step to close the pop over and continue with other steps.

Another way was to `foce` click on elements, but I avoided that. This is just with the test tool. In real time, this works fine.

**Comment by yasserfaraazkhan** [comment]
Sure  @streamer45 
I'll make a note in Jira to add a step based on the fix pushed.

Currently, we do have a step that checks if the button is disabled and it is passing.
https://github.com/mattermost/mattermost-plugin-calls/blob/main/e2e/tests/join_call.spec.ts#L129

please let me know, if you'd suggest to merge or put a pin on this until fix is ready.

**Comment by streamer45** [comment]
Thanks. I think we can merge this one to fix the pipeline but I'd like to address that eventually because it makes us vulnerable to future regressions.

**Review by streamer45 — APPROVED** [review]

**Comment by yasserfaraazkhan** [comment]
Got it.
Will keep close eye on these tests.

**Pull request merged**